src/override/memcpy.cc

```diff
@@ -193,15 +193,12 @@ namespace
     return (pointer_align_down<Size>(const_cast<void*>(src)) == src) &&
       (pointer_align_down<Size>(dst) == dst);
   }
-}
 
-extern "C"
-{
   /**
    * Snmalloc checked memcpy.
    */
-  SNMALLOC_EXPORT void*
-    SNMALLOC_NAME_MANGLE(memcpy)(void* dst, const void* src, size_t len)
+  template<bool checked = true>
```

Comment:
This probably wants to be in an anonymous namespace, not the snmalloc one, since it shouldn't be exposed outside of this compilation unit.